### ISSUE TITLE
Embed application build information into tracy files

### ISSUE THREAD

**Issue by benvanik** [google/iree#1483]
`TracyAppInfo` can be used to add arbitrary data into the trace. We could record build information in release builds here such that we can correlate saved traces with which revision they were captured at. We don't likely want to do this in dev builds so that we aren't relinking binaries all the time. There may be some cmake magic we could use to make this not so bad.

**Comment by antiagainst** [comment]
Yup. Embedding the SHA from which IREE is built would be quite helpful (at lease before we have version scheme and well-organized releasing cycles.) We used to do that [with DXC](https://github.com/Microsoft/DirectXShaderCompiler/blob/master/docs/SPIR-V.rst#debugging) and allow recording the Git SHA in the generated SPIR-V blob.

**Comment by benvanik** [comment]
Closing as wontfix (unless someone gets motivated). Embedding build information is tricky and we don't really have users following this workflow (today).
